### PR TITLE
Add CBMC proof for IPv6 FreeRTOS_Routing functions

### PR DESCRIPTION
**Pull request by moninom1**
Description
-----------
Add CBMC proof for following functions :
FreeRTOS_MatchingEndpoint
pxEasyFit

Also validates :
xIPv6_GetIPType
FreeRTOS_FirstEndPoint
FreeRTOS_NextEndPoint

Test Steps
-----------
```
python prepare.py
make
```

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [x] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/cbmc/proofs/Routing/MatchingEndpoint/MatchingEndpoint_harness.c [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    ProtocolPacket_t * pxProtocolPacket;
+
+    __CPROVER_assume( pcNetworkBuffer != NULL );
+
+    pxProtocolPacket = ( ProtocolPacket_t * ) ( ( uintptr_t ) ( pcNetworkBuffer ) + 2U );
+    __CPROVER_assume( pxProtocolPacket != NULL );
+
+    /* For this proof, its assumed that the endpoints and interfaces are correctly
+     * initialised and the pointers are set correctly. */
+    pxNetworkEndPoints = ( NetworkEndPoint_t * ) malloc( sizeof( NetworkEndPoint_t ) );
```
Comment (moninom1) [inline test/cbmc/proofs/Routing/MatchingEndpoint/MatchingEndpoint_harness.c:55]:
```suggestion
    pxNetworkEndPoints = ( NetworkEndPoint_t * ) safeMalloc( sizeof( NetworkEndPoint_t ) );
```

File: test/cbmc/proofs/Routing/MatchingEndpoint/Makefile.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{
+    "ENTRY": "MatchingEndpoint",
+    "CBMCFLAGS":
+    [
+      "--unwindset pxEasyFit.0:5",
```
Comment (ActoryOu) [inline test/cbmc/proofs/Routing/MatchingEndpoint/Makefile.json:5]: Can we set a macro to describe what 5 is?

Comment (ActoryOu) [reply test/cbmc/proofs/Routing/MatchingEndpoint/Makefile.json:5]: Like [Aniruddha commented](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/pull/896#discussion_r1230101287) in another PR.

**Review by ActoryOu — CHANGES_REQUESTED** [review]

File: test/cbmc/proofs/Routing/MatchingEndpoint/MatchingEndpoint_harness.c [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    /* For this proof, its assumed that the endpoints and interfaces are correctly
+     * initialised and the pointers are set correctly. */
+    pxNetworkEndPoints = ( NetworkEndPoint_t * ) safeMalloc( sizeof( NetworkEndPoint_t ) );
+    __CPROVER_assume( pxNetworkEndPoints != NULL );
+
+    /* Interface init. */
+    pxNetworkEndPoints->pxNetworkInterface = pxNetworkInterface;
+    __CPROVER_assume( pxNetworkEndPoints->pxNetworkInterface != NULL );
+
+    pxNetworkEndPoints->pxNext = NULL;
```
Comment (ActoryOu) [inline test/cbmc/proofs/Routing/MatchingEndpoint/MatchingEndpoint_harness.c:62]: Should we use nondet_bool to append second endpoint to the list?

**Review by ActoryOu — CHANGES_REQUESTED** [review]

**Review by ActoryOu — APPROVED** [review]

**Review by shubnil — APPROVED** [review]

**Pull request merged**